src/components/common/carousel/carousel.jsx

```diff
@@ -0,0 +1,85 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+
+import { View, Text, ScrollView, TouchableOpacity } from 'react-native';
+import { MAIN_TEXT_COLOR, MAIN_THEME_COLOR } from '../../../constants/theme/colors';
+import styles from './styles';
+
+const Carousel = ({ itemsPerIntervals, data, style }) => {
+  const scrollViewRef = React.createRef();
+  const [width, setWidth] = useState(0);
+  const [currInterval, setCurrInterval] = useState(1);
+
+  const totalItems = data.length;
+  const intervals = Math.ceil(totalItems / itemsPerIntervals);
+
+  const getCurrentInterval = (offset) => {
+    for (let i = 1; i <= intervals; i++) {
+      if (offset < width * i - 0.5 * width) {
```

Comment:
Dzięki temu zapisowi
`- 0.5 * width`
na emulatorze też wszystko działa.
Powoduje to, że porównuje pozycje środka każdego slide, a nie jego końca.

Reply:
komentarz w kodzie a nie na GH xd